### PR TITLE
[Feature] updating Extending TCA examples

### LINKED ISSUE #1236: TCA type=user with userfunction has been deprecated with 9.5
It is still used as example here:
https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/ExtensionArchitecture/ExtendingTca/Examples/Index.html

The example doesn't work in 10  and 11

see: https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/9.5/Deprecation-86163-TCATypeuserWithoutRenderType.html

### PR DESCRIPTION
**Pull request by linawolf**
resolves #1236

Backport to 10.4 and 9.5

references https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-TCA/pull/334

### CODE REVIEW

File: Documentation/ExtensionArchitecture/ExtendingTca/Examples/Index.rst [hunk 1]
```diff
@@ -1,84 +1,95 @@
...
+            'renderType' => 'specialField',
+            'parameters' => [
+               'size' => '30',
+               'color' => '#F49700',
+            ],
+         ],
+      ],
+   ];
+
+   \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addTCAcolumns('fe_users', $temporaryColumns);
```
Comment (mbrodala) [inline Documentation/ExtensionArchitecture/ExtendingTca/Examples/Index.rst:63]: We should directly put the array as 2nd argument here. I never understood why a temporary variable was used if there is no recycling / customization with another call afterwards.

File: Documentation/ExtensionArchitecture/ExtendingTca/Examples/Index.rst [hunk 2]
```diff
@@ -131,39 +145,49 @@ Then we add it to the :php:`$GLOBALS['TCA']` in :file:`Configuration/TCA/Overrid
...
-	$temporaryColumn = array(
-		'tx_examples_noprint' => array (
-			'exclude' => 0,
-			'label' => 'LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:tt_content.tx_examples_noprint',
-			'config' => array (
-				'type' => 'check',
-			)
-		)
-	);
+	$temporaryColumn = [
```
Comment (mbrodala) [inline Documentation/ExtensionArchitecture/ExtendingTca/Examples/Index.rst:148]: Same here.

**Review by mbrodala — CHANGES_REQUESTED** [review]

File: Documentation/ExtensionArchitecture/ExtendingTca/Examples/Index.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/ExtendingTca/Examples/Index.rst]: Did so an rephrased the explanation of the example while doing it.

File: Documentation/ApiOverview/BackendRouting/Index.rst [hunk 3]
```diff
@@ -98,19 +104,50 @@ This technique should be used on all public routes (without session token) that
...
+   
+   
+.. index:: Backend routing; Generating backend URLs
+   
+Generating backend URLs
+=======================
+
+Using the UriBuilder API, you can generate any kind of URL for the Backend, may it be
+a module, a typical route or an AJAX call. Therefore use either :php:`buildUriFromRoute()`
+or :php:`buildUriFromRoutePath()`. The UriBuilder then returns a PSR-7 conform Uri object
```
Comment (mbrodala) [inline Documentation/ApiOverview/BackendRouting/Index.rst:129]:
```suggestion
or :php:`buildUriFromRoutePath()`. The :php:`UriBuilder` then returns a PSR-7 conform :php:`Uri` object
```

File: Documentation/ApiOverview/BackendRouting/Index.rst [hunk 4]
```diff
@@ -98,19 +104,50 @@ This technique should be used on all public routes (without session token) that
...
+that can be casted to string when needed. Furthermore does the UriBuilder automatically
```
Comment (mbrodala) [inline Documentation/ApiOverview/BackendRouting/Index.rst:130]:
```suggestion
that can be cast to string when needed. Furthermore does the :php:`UriBuilder` automatically
```

File: Documentation/ApiOverview/BackendRouting/Index.rst [hunk 5]
```diff
@@ -98,19 +104,50 @@ This technique should be used on all public routes (without session token) that
...
+generate and apply the mentioned session token.
```
Comment (mbrodala) [inline Documentation/ApiOverview/BackendRouting/Index.rst:131]:
```suggestion
generates and applies the mentioned session token.
```

File: Documentation/ApiOverview/BackendRouting/Index.rst [hunk 6]
```diff
@@ -98,19 +104,50 @@ This technique should be used on all public routes (without session token) that
...
+
+.. code-block:: php
+
+   // Using a route identifier
+   $uriBuilder = GeneralUtility::makeInstance(UriBuilder::class);
```
Comment (mbrodala) [inline Documentation/ApiOverview/BackendRouting/Index.rst:136]: We should mention/link DI here IMO.

**Review by mbrodala — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/BackendRouting/Index.rst (hunk 6 above)
Comment (linawolf) [reply Documentation/ApiOverview/BackendRouting/Index.rst:136]: What is "DI"?

**Comment by linawolf** [comment]
@mbrodala can we merge this?

File: Documentation/ApiOverview/BackendRouting/Index.rst (hunk 6 above)
Comment (mbrodala) [reply Documentation/ApiOverview/BackendRouting/Index.rst:136]: Dependency Injection: https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/ApiOverview/DependencyInjection/Index.html

Comment (linawolf) [reply Documentation/ApiOverview/BackendRouting/Index.rst:136]: This change just crept in here from rebasing. It was not part of this PR originally. So I would suggest you open an issue or PR about this and suggest a change.

**Comment by linawolf** [comment]
I am merging this now before we get new conflicts. Please open new PRs or issues for further changes

**Pull request merged**